test/brotli.spec.ts

```diff
@@ -68,4 +68,151 @@ describe("Brotli-wasm", () => {
+        expect(encStream.result()).to.equal(brotli.BrotliStreamResult.NeedsMoreInput);
+        const encOutput2 = encStream.compress(encInput2, 100);
+        expect(encStream.result()).to.equal(brotli.BrotliStreamResult.NeedsMoreInput);
+        const encOutput3 = encStream.compress(undefined, 100);
+        expect(encStream.result()).to.equal(brotli.BrotliStreamResult.ResultSuccess);
+        const encOutput = Buffer.concat([encOutput1, encOutput2, encOutput3]);
+
+        const decInput1 = encOutput.slice(0, encOutput.length / 2);
+        const decInput2 = encOutput.slice(encOutput.length / 2);
+        const decStream = new brotli.DecompressStream();
+        const decOutput1 = decStream.decompress(decInput1, 100);
+        expect(decStream.result()).to.equal(brotli.BrotliStreamResult.NeedsMoreInput);
+        const decOutput2 = decStream.decompress(decInput2, 100);
+        expect(decStream.result()).to.equal(brotli.BrotliStreamResult.ResultSuccess);
+        const decOutput = Buffer.concat([decOutput1, decOutput2]);
+
+        expect(decOutput.toString('utf8')).to.equal(s);
+    });
+
+    it("can compress & decompress back to the original result with a different quality setting", () => {
+        const s = "Some thrilling text I urgently need to compress";
+        const encInput = Buffer.from(s);
+        const encInput1 = encInput.slice(0, encInput.length / 2);
+        const encInput2 = encInput.slice(encInput.length / 2);
+        const quality = 3;
+        const encStream = new brotli.CompressStream(quality);
+        const encOutput1 = encStream.compress(encInput1, 100);
+        expect(encStream.result()).to.equal(brotli.BrotliStreamResult.NeedsMoreInput);
+        const encOutput2 = encStream.compress(encInput2, 100);
+        expect(encStream.result()).to.equal(brotli.BrotliStreamResult.NeedsMoreInput);
+        const encOutput3 = encStream.compress(undefined, 100);
+        expect(encStream.result()).to.equal(brotli.BrotliStreamResult.ResultSuccess);
+        const encOutput = Buffer.concat([encOutput1, encOutput2, encOutput3]);
+
+        const decInput1 = encOutput.slice(0, encOutput.length / 2);
+        const decInput2 = encOutput.slice(encOutput.length / 2);
+        const decStream = new brotli.DecompressStream();
+        const decOutput1 = decStream.decompress(decInput1, 100);
+        expect(decStream.result()).to.equal(brotli.BrotliStreamResult.NeedsMoreInput);
+        const decOutput2 = decStream.decompress(decInput2, 100);
+        expect(decStream.result()).to.equal(brotli.BrotliStreamResult.ResultSuccess);
+        const decOutput = Buffer.concat([decOutput1, decOutput2]);
+
+        expect(decOutput.toString('utf8')).to.equal(s);
+    });
+
+    it("streaming compressing can handle needing more output", () => {
+        // Suppressed currently since it requires a large file as input
+        return;
+        // const input = fs.readFileSync('test/encLongInput.bin');
```

Comment:
How about we generate a large buffer here directly in memory? You can use `Buffer.alloc(size, "hello world")` to create a buffer of any arbitrary size that just repeats 'hello world' over and over again.

Reply:
Repeating strings will be compressed to a short output by Brotli. To make the output long enough, the input should have enough randomness to make Brotli Huffman Tree inefficient.

Now I have modified it to generate random bytes in code. It costs a relatively long time (about 1s) to run the test case but should be fine personally speaking.